app/controllers/api/v3x1/sources_controller.rb

```diff
@@ -0,0 +1,17 @@
+module Api
+  module V3x1
+    class SourcesController < Api::V3x0::SourcesController
+      def destroy
+        source = Source.find(params.require(:id)).tap { |s| authorize(s) }
+
+        if source.super_key?
```

Comment:
it's not a subject of this PR, but I can image more descriptive method name, i.e. `source.has_super_key?` ot `source.super_key_auth?`